src/components/scenes/CoinRankingScene.tsx

```diff
@@ -0,0 +1,290 @@
+import { FlatListItem } from '../../types/types'
+import { debugLog, enableDebugLogType, LOG_COINRANK } from '../../util/logger'
+import { fetchRates } from '../../util/network'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { CoinRankRow } from '../data/row/CoinRankRow'
+import { showError } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { DividerLine } from '../themed/DividerLine'
+import { EdgeText } from '../themed/EdgeText'
+import { OutlinedTextInput, OutlinedTextInputRef } from '../themed/OutlinedTextInput'
+
+const coinRowData: CoinRowData = { coinRankings: [] }
+
+const QUERY_PAGE_SIZE = 30
+const LISTINGS_REFRESH_INTERVAL = 30000
+
+// Masking enable bit with 0 disables logging
+enableDebugLogType(LOG_COINRANK & 0)
+
+interface Props {
+  // navigation: NavigationProp<'coinRanking'>
+  // route: RouteProp<'coinRanking'>
+}
+
+const percentChangeOrder: PercentChangeTimeFrame[] = ['hours1', 'hours24', 'days7', 'days30', 'year1']
+const percentChangeStrings: { [pc: string]: string } = {
+  hours1: '1hr',
+  hours24: '24hr',
+  days7: '7d',
+  days30: '30d',
+  year1: '1y'
+}
+const priceSubTextStrings: { [pc: string]: string } = {
+  marketCap: 'MCap',
+  volume24h: 'Vol 24h'
+}
+
+type Timeout = ReturnType<typeof setTimeout>
+const CoinRankingComponent = (props: Props) => {
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const mounted = React.useRef<boolean>(true)
+  const textInput = React.useRef<OutlinedTextInputRef>(null)
+  const timeoutHandler = React.useRef<Timeout | undefined>()
+
+  const [requestDataSize, setRequestDataSize] = useState<number>(QUERY_PAGE_SIZE)
+  const [dataSize, setDataSize] = useState<number>(0)
+  const [searchText, setSearchText] = useState<string>('')
+  const [searching, setSearching] = useState<boolean>(false)
+  const [percentChangeTimeFrame, setPercentChangeTimeFrame] = useState<PercentChangeTimeFrame>('hours24')
+  const [priceSubText, setPriceSubText] = useState<PriceSubText>('marketCap')
+  const [fiatCode] = useState<string>('iso:USD')
+
+  const { coinRankings } = coinRowData
+
+  const renderItem = useHandler((itemObj: FlatListItem<number>) => {
+    const { index, item } = itemObj
+    const currencyCode = coinRankings[index]?.currencyCode ?? 'NO_CURRENCY_CODE'
+    const rank = coinRankings[index]?.rank ?? 'NO_RANK'
+    const key = `${index}-${item}-${rank}-${currencyCode}`
+    debugLog(LOG_COINRANK, `renderItem ${key.toString()}`)
+
+    return <CoinRankRow index={item} key={key} coinRowData={coinRowData} percentChangeTimeFrame={percentChangeTimeFrame} priceSubText={priceSubText} />
+  })
+
+  const handleEndReached = useHandler(() => {
+    debugLog(LOG_COINRANK, `handleEndReached. setRequestDataSize ${requestDataSize + QUERY_PAGE_SIZE}`)
+    setRequestDataSize(requestDataSize + QUERY_PAGE_SIZE)
+  })
+
+  const handlePercentChange = useHandler(() => {
```

Comment:
This control challenges common assumptions. Tapping a column title without any other affordances I would expect to sort the list based on the column. 

I understand that we have a convention in our app that certain colors are tappable, and colors are good as secondary indicators, but colors alone are bad at conveying information.

I propose following a more common convention such as the one shown on CoinGecko's app with two rows

1. A radio dropdown when tapping this title allowing the user to select the timeframe, along with a visual down triangle indicator there are more options on the first row:
7d▼

2. The column title on the 2nd row along with an indicator displaying the sort status to the right of the column title, either up triangle, down triangle, or both (ascending sort, descending sort, none).
7d△